common/containers/Tabs/Swap/index.js

```diff
@@ -0,0 +1,120 @@
+        originKindOptions: PropTypes.array,
+        swapOriginKindTo: PropTypes.func,
+        swapDestinationKindTo: PropTypes.func,
+        swapOriginAmountTo: PropTypes.func,
+        swapDestinationAmountTo: PropTypes.func,
+        swapUpdateBityRatesTo: PropTypes.func
+
+    };
+
+    render() {
+        let {
+            bityRates,
+            originAmount,
+            destinationAmount,
+            originKind,
+            destinationKind,
+            destinationKindOptions,
+            originKindOptions,
+            swapOriginKindTo,
+            swapDestinationKindTo,
+            swapOriginAmountTo,
+            swapDestinationAmountTo
+        } = this.props;
+
+        let wantToSwapMyProps = {
+            bityRates,
+            originAmount,
+            destinationAmount,
+            originKind,
+            destinationKind,
+            destinationKindOptions,
+            originKindOptions,
+            swapOriginKindTo,
+            swapDestinationKindTo,
+            swapOriginAmountTo,
+            swapDestinationAmountTo
+        };
+
+
+        if (!bityRates.ETHBTC || !bityRates.ETHREP || !bityRates.BTCETH || !bityRates.BTCREP) {
+            this.bity.getAllRates()
+                .then((data) => {
+                    this.props.swapUpdateBityRatesTo(data);
+                })
+        }
+
+        return (
+            <section className="container" style={{minHeight: '50%'}}>
+                <div className="tab-content">
+                    <main className="tab-pane swap-tab">
+                        <CurrentRates {...bityRates}/>
+                        <WantToSwapMy {...wantToSwapMyProps}/>
+                    </main>
+                </div>
+            </section>
+        )
+    }
+}
+
+function mapStateToProps(state) {
+    return {
+        originAmount: state.swap.originAmount,
+        destinationAmount: state.swap.destinationAmount,
+        originKind: state.swap.originKind,
+        destinationKind: state.swap.destinationKind,
+        destinationKindOptions: state.swap.destinationKindOptions,
+        originKindOptions: state.swap.originKindOptions,
+        bityRates: state.swap.bityRates
+    }
+}
+
+function mapDispatchToProps(dispatch) {
+    return {
+        swapOriginKindTo: (originValue) => {
+            dispatch(SWAP_ORIGIN_KIND_TO(originValue))
+        },
+        swapDestinationKindTo: (destinationValue) => {
+            dispatch(SWAP_DESTINATION_KIND_TO(destinationValue))
+        },
+        swapOriginAmountTo: (originAmount) => {
+            dispatch(SWAP_ORIGIN_AMOUNT_TO(originAmount))
+        },
+        swapDestinationAmountTo: (destinationValue) => {
+            dispatch(SWAP_DESTINATION_AMOUNT_TO(destinationValue))
+        },
+        swapUpdateBityRatesTo: (bityRates) => {
+            dispatch(SWAP_UPDATE_BITY_RATES_TO(bityRates))
+        }
+    }
+}
+
+export default connect(mapStateToProps, mapDispatchToProps)(Swap)
```

Comment:
I'm curious as to why we do this. Is it just to transform the `state.swap.originAmount` => `state.originAmount`?

If that's the reason, I think we should just pass in the props we are expecting at a higher level <Swap {...props.swap} />

Maybe I'm misunderstanding this, let me know.